File: back/api/schemas/holding.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class HoldingCreate(BaseModel):
    name: str
    consultant_id: uuid.UUID | None = None


class HoldingOut(BaseModel):
    id: uuid.UUID
    name: str
    status: str
    client_id: uuid.UUID
    consultant_id: uuid.UUID | None
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True


class HoldingStageOut(BaseModel):
    id: uuid.UUID
    name: str
    description: str
    order: int
    status: str
    stage_type: str
    start_date: datetime | None
    end_date: datetime | None

    class Config:
        orm_mode = True

File: front/app/types/holding.ts
export interface Holding {
  id: string
  name: string
  status: string
  client_id: string
  consultant_id?: string
  created_at: string
  updated_at: string
}

export interface HoldingStage {
  id: string
  name: string
  description: string
  order: number
  status: string
  stage_type: string
  start_date?: string
  end_date?: string
}

export interface DocumentRequirement {
  id: string
  name: string
  description: string
  doc_type: string
  is_required: boolean
  status?: string
}
